### PR TITLE
refactor!(core): remove `Icon` enum and add `Image` type

### PR DESCRIPTION
**Pull request by amrbashir**

### CODE REVIEW

**Comment by amrbashir** [comment]
I will add the equivalent JS types later on another PR

File: .changes/runtime-icon-lifetime.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'tauri-runtime': 'major:breaking'
+---
+
+Add a lifetime paramter for `Icon` type. Also changed `rgba` field to be `Cow<'a, [u8]>`
```
Comment (lucasfernog) [inline .changes/runtime-icon-lifetime.md]:
```suggestion
Add a lifetime parameter for `Icon` type. Also changed `rgba` field to be `Cow<'a, [u8]>`
```

File: .changes/ico-featrue-flags.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'tauri': 'major:breaking'
+---
+
+Renamed `icon-ico` and `icon-png` feature flags, to `image-png` and `image-ico`
```
Comment (lucasfernog) [inline .changes/ico-featrue-flags.md]:
```suggestion
Renamed `icon-ico` and `icon-png` feature flags to `image-ico` and `image-png` respectively
```

File: core/tauri/src/image.rs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+/// An RGBA Image in row-major order from top to bottom.
+#[derive(Debug, Clone)]
+pub struct Image<'a> {
+  rgba: Cow<'a, [u8]>,
+  width: u32,
+  height: u32,
+}
+
+impl<'a> Image<'a> {
+  /// Creates a new Image using RGBA data, in row-major order from top to bottom, and with specified height and width.
```
Comment (lucasfernog) [inline core/tauri/src/image.rs]:
```suggestion
  /// Creates a new Image using RGBA data, in row-major order from top to bottom, and with specified width and height.
```

File: core/tauri/src/image.rs [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+  fn try_from(img: Image<'_>) -> Result<Self, Self::Error> {
+    tray_icon::Icon::from_rgba(img.rgba.to_vec(), img.width, img.height).map_err(Into::into)
+  }
+}
+
+#[cfg(desktop)]
+#[derive(serde::Deserialize)]
+#[serde(untagged)]
+pub enum JsIcon<'a> {
```
Comment (lucasfernog) [inline core/tauri/src/image.rs:159]: why isn't this called JsImage?

**Review by lucasfernog — CHANGES_REQUESTED** [review]

File: core/tauri/src/image.rs (hunk 4 above)
Comment (amrbashir) [reply core/tauri/src/image.rs:159]: Have no idea tbh 😅, I will open another PR for JS implementation and will refactor it later.

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**